templates/tower-project.yaml

```diff
@@ -0,0 +1,368 @@
+
+  LifecycleDataExpiration:
+    Type: Number
+    Description: >-
+      Number of days (from creation) when objects are
+      deleted from S3 and the LifecycleDataStorageClass
+    Default: 365000
+
+  AdminIds:
+    Type: List<String>
+    Description: >-
+      List of IAM user and role IDs (strings) with administrator
+      access and should retain access to restricted resources
+      (e.g., private S3 bucket, KMS key, secrets). The role ID
+      can be obtained in one of two ways.
+
+      1) You can assume the role in question and run the following
+      command, and the role ID is the substring before the colon.
+
+      aws sts get-caller-identity --output text --query UserId
+
+      2) With the appropriate IAM permissions (with an admin role),
+      you can run the following command, where `<role_name>` corresponds
+      to the role name (see examples below to extract the role name).
+
+      aws iam get-role --role-name <role_name> --output text --query Role.RoleId
+
+      For example, if the role ARN is
+      arn:aws:iam::456789012345:role/aws-reserved/sso.amazonaws.com/AWSReservedSSO_Viewer_7e6c15a6e90908df
+      the role name is AWSReservedSSO_Viewer_7e6c15a6e90908df.
+      If you have the ARN for an assumed role (distinguished by the
+      `arn:aws:sts::` prefix and `assumed-role/` substring), you need
+      to exclude the session name at the end of the ARN. For instance,
+      the following assumed-role ARN has the same role name as above:
+      arn:aws:sts::035458030717:assumed-role/AWSReservedSSO_Viewer_fd80909e6a51c6e7/bruno.grande@sagebase.org.
+
+  TemplateRootUrl:
+    Type: String
+    Description: URL of S3 bucket where templates are deployed
+    ConstraintDescription: Must be a valid S3 HTTP URL
+
+  TowerForgePolicyArn:
+    Type: String
+    Description: ARN for Tower Forge IAM policy
+
+  TowerLaunchPolicyArn:
+    Type: String
+    Description: ARN for Tower Launch IAM policy
+
+Conditions:
+  HasReadWriteAccessArns:
+    !Not [!Equals [!Join [",", !Ref ReadWriteAccessArns], ""]]
+  HasReadOnlyAccessArns:
+    !Not [!Equals [!Join [",", !Ref ReadOnlyAccessArns], ""]]
+  HasAnyAccessArns:
+    !Or [!Condition HasReadWriteAccessArns, !Condition HasReadOnlyAccessArns]
+
+Resources:
+  ReadWriteAccessRole:
+    Type: AWS::IAM::Role
+    Condition: HasReadWriteAccessArns
+    Properties:
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonS3FullAccess
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              AWS:
+                !Split
+                  - "|"
+                  - !Join
+                    - "|"
+                    - - !Join
+                        - "|"
+                        - !Ref ReadWriteAccessArns
+                      - !Join
+                        - "|"
+                        - !Ref AdminIds
+            Action:
+              - "sts:AssumeRole"
```

Comment:
hmm, i'm not sure i like this idea.  This allows JC viewers to assume a role that provides full S3 access.  This seems like the same security problem that we discussed in slack.  For example when SageIT assigns a JC user to a nextflow viewer group we expect that they only have viewer access.  If you do this then that expectation becomes irrelevant.  Basically our JC configurations will have no meaning.  The source of truth for access should be JC user groups.

Reply:
We don't want the JC role on the production account to have anything more than `ViewOnlyAccess`, but we also need the ability to grant specific individuals with write-access to the resources they deployed using this template. 

Admittedly, granting write-access to a role called `Viewer` is counter-intuitive. To avoid any surprises, I propose that we can get rid of the `Viewer` role on our production account and deploy the `TowerUser` role described in Sage-Bionetworks-IT/organizations-infra#278. I updated the [role description](https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/278/files#diff-c89df69681d7714e12cc8907e3d7e5b10a9bee477bbff86cd269b81ad0c7a1b5R519-R521) so that the intended use is clear:

> SSO: View-only role that is also used for assuming project-specific roles with additional access

This way, it's clear to Sage IT that adding a user to the `aws-workflow-nextflow-tower-user` JC group not only grant view-access, but also the opportunity to assume a separate role with additional resource-specific write-access. 

If you think this is a fair approach, you can go ahead and approve this PR and Sage-Bionetworks-IT/organizations-infra#278.

Reply:
I like this approach